src/PrestaShopBundle/Resources/config/services/bundle/new_api.yml

```
@@ -0,0 +1,64 @@
+services:
+  _defaults:
+    public: true
+
+  prestashop.api.repository.client:
+    class: PrestaShopBundle\Security\OAuth2\Repository\ClientRepository
+    arguments:
+      - '@security.user.provider.concrete.oauth2'
+      - '@security.user_password_encoder.generic'
+
+  prestashop.api.repository.access_token:
+    class: PrestaShopBundle\Security\OAuth2\Repository\AccessTokenRepository
+
+  prestashop.api.repository.scope:
+    class: PrestaShopBundle\Security\OAuth2\Repository\NullScopeRepository
+
+  prestashop.api.grant.client_credentials:
+    class: League\OAuth2\Server\Grant\ClientCredentialsGrant
+
+  prestashop.api.private_key:
+    class: League\OAuth2\Server\CryptKey
+    arguments: [ '%api_private_key%' ]
+
+  prestashop.api.public_key:
+    class: League\OAuth2\Server\CryptKey
+    arguments: [ '%api_public_key%' ]
+
+  prestashop.api.authorization_server:
+    class: League\OAuth2\Server\AuthorizationServer
+    arguments:
+      - '@prestashop.api.repository.client'
+      - '@prestashop.api.repository.access_token'
+      - '@prestashop.api.repository.scope'
+      - '@prestashop.api.private_key'
+      - '%new_cookie_key%'
+    calls:
+      - [ 'enableGrantType', [ '@prestashop.api.grant.client_credentials' ] ]
+
+  prestashop.api.resource_server:
+    class: League\OAuth2\Server\ResourceServer
+    arguments:
+      - '@prestashop.api.repository.access_token'
+      - '@prestashop.api.public_key'
+
+  prestashop.api.factory.psr17:
+    class: Nyholm\Psr7\Factory\Psr17Factory
+
+  prestashop.api.factory.psrHttpFactory:
```

```suggestion
  prestashop.api.factory.psr_http_factory:
```
We don't use camel case in the service names